azext_iot/central/providers/preview/device_provider_preview.py

```diff
@@ -0,0 +1,181 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from knack.log import get_logger
+from azext_iot.constants import CENTRAL_ENDPOINT
+from azext_iot.central import services as central_services
+from azext_iot.central.models.enum import DeviceStatus, ApiVersion
+
+
+logger = get_logger(__name__)
+
+
+class CentralDeviceProviderPreview:
+    def __init__(self, cmd, app_id: str, token=None):
+        """
+        Provider for device APIs
+
+        Args:
+            cmd: command passed into az
+            app_id: name of app (used for forming request URL)
+            token: (OPTIONAL) authorization token to fetch device details from IoTC.
+                MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+                Useful in scenarios where user doesn't own the app
+                therefore AAD token won't work, but a SAS token generated by owner will
+        """
+        self._cmd = cmd
+        self._app_id = app_id
+        self._token = token
+        self._devices = {}
+        self._device_templates = {}
+        self._device_credentials = {}
+        self._device_registration_info = {}
+
+    def get_device(
+        self, device_id, central_dns_suffix=CENTRAL_ENDPOINT,
+    ):
+        if not device_id:
```

Comment:
device_id is a required parameter at the command bound function layer, in practical usage is this guard for a falsey device_id that useful? Also there is not the same check for get_device_template in the template provider.

![image](https://user-images.githubusercontent.com/28658112/118057752-68f40200-b341-11eb-88f9-1de4a4372d8d.png)

Reply:
fixed - removed